.github/workflows/build.yml

```diff
@@ -120,98 +144,222 @@ jobs:
+            type=semver,pattern={{major}}.{{minor}},priority=10
+            type=semver,pattern={{version}},priority=20
+            type=ref,event=branch,priority=30
+            type=sha,priority=40
+
+      - name: Build circ-scripts image
+        id: build-scripts
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ./docker/Dockerfile
+          target: scripts
+          build-args: |
+            BASE_IMAGE=${{ steps.baseimage.outputs.tag }}
+          labels: ${{ steps.meta-scripts.outputs.labels }}
+          outputs: type=image,"name=${{ steps.ghcr-repo.outputs.scripts }}",push-by-digest=true,name-canonical=true,push=true
+
+      - name: Generate tags for circ-exec
+        id: meta-exec
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ steps.ghcr-repo.outputs.exec }}
+          tags: |
+            type=semver,pattern={{major}}.{{minor}},priority=10
+            type=semver,pattern={{version}},priority=20
+            type=ref,event=branch,priority=30
+            type=sha,priority=40
+
+      - name: Build circ-exec image
+        id: build-exec
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ./docker/Dockerfile
+          target: exec
+          build-args: |
+            BASE_IMAGE=${{ steps.baseimage.outputs.tag }}
+          labels: ${{ steps.meta-exec.outputs.labels }}
+          outputs: type=image,"name=${{ steps.ghcr-repo.outputs.exec }}",push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digests
+        run: |
+          mkdir -p ${{ runner.temp }}/digests/webapp
+          webapp_digest="${{ steps.build-webapp.outputs.digest }}"
+          touch "${{ runner.temp }}/digests/webapp/${webapp_digest#sha256:}"
+          echo "WEBAPP_DIGEST=$webapp_digest"
+          mkdir -p ${{ runner.temp }}/digests/scripts
+          scripts_digest="${{ steps.build-scripts.outputs.digest }}"
+          touch "${{ runner.temp }}/digests/scripts/${scripts_digest#sha256:}"
+          echo "SCRIPTS_DIGEST=$scripts_digest"
+          mkdir -p ${{ runner.temp }}/digests/exec
+          exec_digest="${{ steps.build-exec.outputs.digest }}"
+          touch "${{ runner.temp }}/digests/exec/${exec_digest#sha256:}"
+          echo "EXEC_DIGEST=$exec_digest"
+
+      - name: Upload digests
+        uses: actions/upload-artifact@v4
+        with:
+          name: digests-${{ matrix.arch.name }}
+          path: ${{ runner.temp }}/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
   integration-test:
-    name: Test circ-${{ matrix.image }} (${{ matrix.platform }})
-    runs-on: ubuntu-latest
+    name: Integration test (${{ matrix.arch.name }})
+    runs-on: ${{ matrix.arch.runner }}
     needs: [build]
     permissions:
       contents: read
     strategy:
       fail-fast: false
       matrix:
-        platform: ["linux/amd64", "linux/arm64"]
-        image: ["scripts", "webapp"]
```

Comment:
The majority of the time in this workflow is spent pulling the image, rather then running the tests, so it made sense to combine the job for scripts and webapp, since running the tests themselves is quick.